nolearn/lasagne/visualize.py

```diff
@@ -311,7 +311,12 @@ def make_pydot_graph(layers, output_shape=True, verbose=False):
     pydot_nodes = {}
     pydot_edges = []
     for i, layer in enumerate(layers):
-        layer_type = '{0}'.format(layer.__class__.__name__)
+        layer_name = None
+        if hasattr(layer, 'name'):
+            layer_name = layer.name
+        if layer_name is None:
+            layer_name = layer.__class__.__name__
```

Comment:
Would this work:

```python
layer_name = getattr(layer, 'name', None)
if layer_name is None:
    layer_name = layer.__class__.__name__
```

Reply:
Yes, this would work and it is nicer 👍 
I'll change now :)